src/serial.rs

```diff
@@ -0,0 +1,503 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
nit: Can you make sure that lines are not longer than 100 chars and that we keep the same formatting throughout the file (i.e. comments go on the line before the define and not on the same line as teh define).